Azure-Samples/azure-search-openai-demo
# Add GitHub Actions, move roles to refactored Bicep Files

## Purpose
This PR adds GitHub Actions to the demo repository and fixes a small role assignment issue.

When running the azd deploy multiple times (or GitHub Actions), the user roles addition would crash because the role assignments were named with "subscription+resource group+principal+role" name (see infra\role.bicep line 14). Changed to use "subscription+principal+role" name so this could be deployed multiple times if needed.

As part of this fix, the role definition GUIDs were moved to a JSON file for better readability.

## Does this introduce a breaking change?
```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[x] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
* Get the code
```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
```

* Test the code
```
Set up repository secrets and variables (see .github/workflows/readme.md for details)
```

## What to Check
Verify that the following are valid
* Run the deploy.infra.and.website workflow to test

## Other Information
These examples are set up with 'dev'/'qa'/'prod' GitHub environments as an example, but that could be easily changed.
The workflows are set to be manually triggered for now, not triggered on check-in, but that could be easily changed.
